frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,470 @@
+        />
+      </div>
+    </>
+  );
+};
+
+const OperatorInstallPageView: React.FC<OperatorInstallPageProps> = (props) => {
+  const [loading, setLoading] = React.useState(true);
+  const [installObj, setInstallObj] = React.useState<ClusterServiceVersionKind | InstallPlanKind>();
+  const [subscriptionObj, setSubscriptionObj] = React.useState<SubscriptionKind>();
+  const [status, setStatus] = React.useState('');
+  const [needsApproval, setNeedsApproval] = React.useState(false);
+  const [packageManifest, setPackageManifest] = React.useState<PackageManifestKind>();
+
+  const { resources, targetNamespace, pkgNameWithVersion } = props;
+
+  const isStatusSucceeded = () => {
+    return status === 'Succeeded';
+  };
+  const isStatusFailed = () => {
+    return status === 'Failed';
+  };
+  const isApprovalNeeded = () => {
+    return needsApproval;
+  };
+
+  React.useEffect(() => {
+    if ([...(resources.packageManifest.data as PackageManifestKind[])].length > 0) {
+      setPackageManifest(resources.packageManifest.data[0]);
+    }
+    const tmpCSV = (resources.clusterServiceVersions.data as ClusterServiceVersionKind[]).find(
+      (op) => op.metadata.name === pkgNameWithVersion,
+    ) as ClusterServiceVersionKind;
+    if (typeof tmpCSV !== 'undefined') {
+      const tmpStatus = tmpCSV?.status?.phase || null;
+      if (tmpStatus !== null) {
+        setInstallObj(tmpCSV);
+        setStatus(tmpStatus);
+        setLoading(false);
+      }
+    } else {
+      // There is no ClusterServiceVersion for the package, so look at Subscriptions/InstallPlans
+      const tmpSub = [...(resources.subscription.data as SubscriptionKind[])].find(
+        (op: SubscriptionKind) =>
+          op?.spec?.startingCSV === pkgNameWithVersion && op?.status?.state === 'UpgradePending',
+      ) as SubscriptionKind;
+      if (typeof tmpSub !== 'undefined') {
+        const tmpStatus = tmpSub?.status?.state || null;
+        const ipPlanName = tmpSub?.status?.installplan?.name || '';
+        const tmpInstallPlan: InstallPlanKind = [
+          ...(resources.installPlan.data as InstallPlanKind[]),
+        ].find((ip) => ip.metadata.name === ipPlanName) as InstallPlanKind;
+        setSubscriptionObj(tmpSub);
+        setStatus(tmpStatus);
+        if (typeof tmpInstallPlan !== 'undefined') {
+          setInstallObj(tmpInstallPlan);
+          setNeedsApproval(
+            tmpInstallPlan.spec.approval === 'Manual' && tmpInstallPlan.spec.approved === false,
+          );
+          setLoading(false);
+        }
+      }
+    }
+  }, [loading, pkgNameWithVersion, targetNamespace, resources]);
+
+  const approve = () => {
+    k8sUpdate(InstallPlanModel, { ...installObj, spec: { ...installObj.spec, approved: true } })
+      .then(() => setStatus(''))
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
+  };
+
+  const getIndicator = () => {
```

Comment:
Why is this a function?

```tsx
    let indicator; 
    if (isStatusFailed()) {
      indicator = <RedExclamationCircleIcon size="lg" />;
    }
    // ...

Reply:
Does this function and the one below it cause performance issues?  They are called once per render and they are super simple.  I can switch them to variables, but then the code is not as easily read and is not neatly blocked together.

Reply:
> Does this function and the one below it cause performance issues? They are called once per render and they are super simple. I can switch them to variables, but then the code is not as easily read and is not neatly blocked together.

If you think the component is too large and complicated, it's better to create separate smaller components and compose them. Breaking this into no-arg functions that are called later in render makes the code more complicated, not less imo.